src/components/PaginatedTable/TableChunk.tsx

```diff
@@ -101,7 +103,11 @@ export const TableChunk = typedMemo(function TableChunk<T, F>({
         }
     }, [currentData, isActive, onDataFetched]);
 
-    const dataLength = currentData?.data?.length || limit;
+    const chunkOffset = id * limit;
+    const remainingLenght = totalLength - chunkOffset;
+    const calculatedChunkLength = remainingLenght < limit ? remainingLenght : limit;
```

Comment:
This allows not to show 50 (limit) loading rows for the last chunk, when we know how many entities are left. Also it allows to set some predefined initial length to table